plugins/modules/system/disk_facts.py

```diff
@@ -0,0 +1,181 @@
+    - This module is to get disk facts like free space, used percentage, mount location etc in linux Ansible controller/controlled machines.
+requirements:
+  - psutil(python module)
+options:
+    name:
+        description:
+            - This is the disk name to query
+        required: false
+        default: all
+        type: str
+    filter:
+        description:
+            - This is the parameter used to filter the output
+        required: false
+        type: list
+        elements: str
+author:
+    - Saranya Sridharan(@saranyasridharan), Swetha M, Nagendra Kini, Anup Kumar, Mehmoodkha Pathan
+'''
+
+EXAMPLES = '''
+- name: Get disk facts
+  community.general.disk_facts:
+
+- name: Get Disk facts for Particular disk info
+  community.general.disk_facts:
+    name: diskname
+
+- name: Filter the output parameters for specific disk
+  community.general.disk_facts:
+    name: diskname
+    filter:
+      - mountpoint
+      - freespace
+      - totalsize
+
+- name: Filter the output parameters for all disk
+  community.general.disk_facts:
+    filter:
+      - fstype
+      - usedspace
+      - capacity_percentage
+'''
+
+
+RETURN = '''
+ansible_facts:
+    description: disk information only filtered parameter values if filter option is given.
+    returned: dictionary of disks informations
+    type: dict
+    sample: {
+        '/dev/disk1s1': {
+             'capacity_percentage': '2%'
+        }
+    }
+'''
+
+
+from ansible.module_utils.basic import AnsibleModule
+try:
+    import psutil
+    HAS_PSUTIL = True
+except ImportError:
+    HAS_PSUTIL = False
+
+
+# To calculate the disk size
+def calculate_size(bytes, suffix="B"):
+    """
+    Scale bytes to its proper format
+    e.g:
+        1253656 => '1.20MB'
+        1253656678 => '1.17GB'
+    """
+    factor = 1024
+    for unit in ["", "K", "M", "G", "T", "P"]:
+        if bytes < factor:
+            return "{0:.2f}{1}{2}".format(bytes, unit, suffix)
+        bytes /= factor
+
+
+# To get the disk partitions information
+def disk_facts(devicename):
+    output = {}
+    partitions = psutil.disk_partitions()
+    found = False
+    for partition in partitions:
+        if partition.device == devicename:
+            found = True
+        output[partition.device] = {}
+        output[partition.device]['mountpoint'] = partition.mountpoint
+        output[partition.device]['fstype'] = partition.fstype
+        try:
+            partition_usage = psutil.disk_usage(partition.mountpoint)
+        except PermissionError:
+            # this can be catched due to the disk that
+            # isn't ready
+            continue
+        output[partition.device]['totalsize'] = calculate_size(partition_usage.total)
+        output[partition.device]['usedspace'] = calculate_size(partition_usage.used)
+        output[partition.device]['freespace'] = calculate_size(partition_usage.free)
+        output[partition.device]['capacity_percentage'] = str(partition_usage.percent) + "%"
```

Comment:
Same here: having this as a string makes it pretty much unusable, except for displaying it to a user.